source/testing-your-html/index.html.md.erb

```diff
@@ -0,0 +1,50 @@
+---
+title: Test your HTML matches GOV.UK Frontend
+weight: 61
+---
+
+# Test your HTML matches GOV.UK Frontend
+
+You can use our test fixtures to check you're outputting the same HTML that GOV.UK Frontend uses, when you:
+
+- create your own component macros (also called ‘partials’ or ‘templates’)
+- update GOV.UK Frontend
+
+## Using the HTML test files
+
+If you [installed GOV.UK Frontend with Node.js package manager (npm)](/installing-with-npm), you can find the `fixtures.json` file for each component in the `node_modules/govuk-frontend/govuk/components/COMPONENT-NAME/` folder, where `COMPONENT-NAME` is the name of the component.
+
+<%= warning_text('You should not test your HTML using the YAML files in the <code>src/</code> folder.') %>
+
+For example, you can find the `fixtures.json` file for the button component in the `node_modules/govuk-frontend/govuk/components/button/` folder:
+
+```js
+[
+  {
+    "name": "default",
+    "options": {
+      "text": "Save and continue"
+    },
+    "html": "<button class=\"govuk-button\" data-module=\"govuk-button\">\n  Save and continue\n</button>"
+  },
+  {
+    "name": "secondary button",
+    "options": {
+      "text": "Find address",
+      "classes": "govuk-button--secondary"
+    },
+    "html": "<button class=\"govuk-button govuk-button--secondary\" data-module=\"govuk-button\">\nFind address\n</button>"
+  },
+  ...
+]
+```
+
+Each object in the list is a different example of the component, where:
+
+- `name` is the name of the example
+- `options` are the options that generate this example's `html`
+- `html` is the HTML that GOV.UK Frontend outputs with these options
+
+For each example, pass `options` into your own macro and check the generated HTML matches `html`.
+
+Your HTML may not match exactly. For example, your framework may add extra whitespace or attributes to your HTML.
```

Comment:
@36degrees @vanitabarrett I had a long look at the final paragraph and realised I might be trying to over-explain things. In retrospect, it feels like enough to warn users that the HTML may not match, without going into what to do about it - as this would be fairly self-explanatory? Let me know what you think!

Reply:
I think the word 'may' makes it sound permissive? ('It's OK if the HTML doesn't match exactly')

Do we just need to be clear that they will have to do _something_…? What about something like…

> If your HTML does not match exactly – for example if your framework adds extra whitespace or attributes to your HTML – you'll need to write your tests to allow for known differences.

Reply:
I've had another go in ee2717b - I've mainly used Ollie's suggestion but with slight style tweak. Let me know what you think!